_posts/2018-07-05-kafka-nfs.md

```diff
@@ -0,0 +1,35 @@
+---
+layout:     post
+title:      Kafka on NFS
+date:       2018-07-05
+summary:    There is a general recommendation against running Apache Kafka on NFS storage, but nobody really gives a good explanation as to why.  In this post we look at some broker crashes we have seen happening on Kafka clusters which use NFS storage and why they were happening.
+author:     alice_kaerast
+image:      kafka-nfs/before-reallocation.png
+category:   Operations
+---
+
+There is a general recommendation against running Apache Kafka on NFS storage.  But the Jira tickets relating to this are closed resolved, so why is this still a recommendation?  Surely using NFS on a nice big NAS that our storage team provide will make our Kafka cluster for log shipping nice and easy to maintain.  The logs get queued locally if Kafka is down, performance isn't a huge concern for this cluster, and we didn't know how much this cluster would grow over time.  So using NFS exports on the NAS server that our storage team provide seemed like an easy win - cheap storage, easily expandable in the future.
```

Comment:
Repetition of `NAS server that our storage team provide` feels a little odd

Reply:
Yeah, I wasn't sure if that was too much repetition.  I've rewritten this bit now.